lib/gitlib.sh

```diff
@@ -41,7 +41,7 @@ BRANCH_REGEX="master|release-([0-9]{1,})\.([0-9]{1,})(\.([0-9]{1,}))*$"
 # build - 1=build number, 2=sha1
 declare -A VER_REGEX=([release]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-alpha|-beta)*\.*(0|[1-9][0-9]*)?"
                       [dotzero]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.0$"
-                      [build]="([0-9]{1,})\+([0-9a-f]{5,40})"
+                      [build]="([0-9]{1,})\+([0-9a-f]{5,40})(-dirty)?"
```

Comment:
this is used in places like `anago`, where we definitely don't want to match `-dirty`.